src/encoded/static/components/cart/cart.js

```diff
@@ -411,65 +683,164 @@ class FileFacets extends React.Component {
+                    // allowing the user to select it to extend the set of selected files.
+                    selectedFacetKeys.forEach((selectedFacetField) => {
+                        // Remove one facet containing a selection from the test.
+                        const filteredSelectedFacetKeys = selectedFacetKeys.filter(key => key !== selectedFacetField);
+                        match = filteredSelectedFacetKeys.every((filteredSelectedFacetKey) => {
+                            const filePropValue = getObjectFieldValue(file, filteredSelectedFacetKey);
+                            return selectedTerms[filteredSelectedFacetKey].indexOf(filePropValue) !== -1;
+                        });
 
-        // Detect if we have empty facets.
-        let emptyFacets = false;
-        if (displayedFacets) {
-            emptyFacets = !Object.keys(displayedFacets).some(field => displayedFacets[field].terms && displayedFacets[field].terms.length > 0);
+                        // A match means to add to the count of the current facet field file term
+                        // only.
+                        if (match) {
+                            FileFacets.addFileTermToFacet(facets, selectedFacetField, file);
+                        }
+                    });
+                }
+            });
+
+            // We need to include selected terms that happen to have a zero count, so add all
+            // selected facet terms not yet included in `facets`.
+            Object.keys(selectedTerms).forEach((field) => {
+                if (field in facets) {
+                    // Find selected terms NOT in facets and add them with a zero count.
+                    const missingTerms = selectedTerms[field].filter(term => Object.keys(facets[field]).indexOf(term) === -1);
+                    if (missingTerms.length > 0) {
+                        missingTerms.forEach((term) => {
+                            facets[field][term] = 0;
+                        });
+                    }
+                }
+            });
         }
 
+        return { facets: Object.keys(facets).length > 0 ? facets : null, selectedFileCount };
+    }
+
+    /**
+     * Called when the user clicks a facet expander button. Updates the expander states so the
+     * facets re-render to the new expanded states.
+     * @param {string} field Field name for clicked facet expander
+     * @param {bool}   altKey True if alt/option key down at time of click
+     */
+    handleExpanderClick(field, altKey) {
+        this.setState((state) => {
+            if (altKey) {
+                // Alt key held down, so expand or collapse *all* facets.
+                const nextExpandedState = !state.expanded[field];
+                const expandedStates = {};
+                displayedFacetFields.forEach((facetField) => {
+                    expandedStates[facetField] = nextExpandedState;
+                });
+                return { expanded: expandedStates };
+            }
```

Comment:
This is cool! When we build the advanced facet search we should do this too. And possibly this should also be implemented for the FAQ.